### PR TITLE
Setup database

### PR DESCRIPTION
**Pull request by ThunderDev1**
Setup the database and added a few routes:
- create a game
- add a player
- get game details

### CODE REVIEW

**Comment by bellangerq** [comment]
Add documentation on how to install .NET packages.

File: server/DTO/GameDTOs.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    // game options would go here
+  }
+
+  public class GameDTO
+  {
+    public GameDTO(Game game)
+    {
+      id = game.Id;
+      hostId = game.HostId;
+      status = game.Status.GetHashCode();
```
Comment (hissalht) [inline server/DTO/GameDTOs.cs]: Could we have some string enums for the game status ?

Comment (hissalht) [reply server/DTO/GameDTOs.cs]: Just for the JSON response, it would be more legible

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      }
+    },
+    {
+      "name": ".NET Core Attach",
+      "type": "coreclr",
+      "request": "attach",
+      "processId": "${command:pickProcess}"
+    }
+  ]
+}
```
Comment (hissalht) [inline .vscode/launch.json]: Not sure that text editor configs should be versioned

File: .vscode/tasks.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                "watch",
+                "run",
+                "${workspaceFolder}/server/server.csproj",
+                "/property:GenerateFullPaths=true",
+                "/consoleloggerparameters:NoSummary"
+            ],
+            "problemMatcher": "$msCompile"
+        }
+    ]
+}
```
Comment (hissalht) [inline .vscode/tasks.json]: same as for `.vscode/launch.json`

File: server/Readme.md [hunk 4]
```diff
@@ -6,10 +6,74 @@ This project is a REST Api made with dotnet core 3.0
...
+├── Common/ # Shared logic.
+├── DTO/ # Data models exposed from the api.
+├── Controllers/ # Api endpoint definition.
+├── Data/ # Database related code.
+|   └── Entities/ # Classes used to create/interact with the database.
+└── Migrations/ # Entity framework generated code for updating the database.
+└── Services/ # Business logic (eg. database calls).
+```
+
+## Routes (WIP)
```
Comment (hissalht) [inline server/Readme.md]: could you add some kind of naming conventions for future routes ?

`/room/1234/player/abcd` vs `/rooms/1234/players/abcde`

Comment (ThunderDev1) [reply server/Readme.md]: ok, using this https://restfulapi.net/resource-naming/

**Pull request merged**